### PR TITLE
Feature/venom inline l3

### PR DESCRIPTION
**Pull request by nqb**
# Description
Add inline L3 capabilities to virtual environment used for integration tests

# Impacts
Integration tests

# Delete branch after merge
YES

# Checklist
(REQUIRED) - [yes, no or n/a]
- [X] Document the feature

### CODE REVIEW

File: addons/vagrant/cumulus/Vagrantfile [hunk 1]
```diff
@@ -361,4 +387,111 @@ vagrant_interface_rule
...
+
+    # Install Rules for the interface re-map
+    device.vm.provision :shell , :inline => <<-delete_udev_directory
+if [ -d "/etc/udev/rules.d/70-persistent-net.rules" ]; then
+    rm -rfv /etc/udev/rules.d/70-persistent-net.rules &> /dev/null
+fi
+rm -rfv /etc/udev/rules.d/70-persistent-net.rules &> /dev/null
+delete_udev_directory
+
+device.vm.provision :shell , :inline => <<-udev_rule
```
Comment (JeGoi) [inline addons/vagrant/cumulus/Vagrantfile:464]: Indentation is missing.

File: addons/vagrant/cumulus/Vagrantfile [hunk 2]
```diff
@@ -361,4 +387,111 @@ vagrant_interface_rule
...
+    device.vm.provision :shell , privileged: false, :inline => 'echo "$(whoami)" > /tmp/normal_user'
+    device.vm.provision "config_node", type: "shell", path: "./helper_scripts/config_node02.sh"
+
+
+    # Install Rules for the interface re-map
+    device.vm.provision :shell , :inline => <<-delete_udev_directory
+if [ -d "/etc/udev/rules.d/70-persistent-net.rules" ]; then
+    rm -rfv /etc/udev/rules.d/70-persistent-net.rules &> /dev/null
+fi
+rm -rfv /etc/udev/rules.d/70-persistent-net.rules &> /dev/null
```
Comment (JeGoi) [inline addons/vagrant/cumulus/Vagrantfile:461]: Remove this line or the statement before

Comment (JeGoi) [inline addons/vagrant/cumulus/Vagrantfile]:
```suggestion

```

File: addons/vagrant/cumulus/Vagrantfile (hunk 1 above)
Comment (JeGoi) [inline addons/vagrant/cumulus/Vagrantfile]:
```suggestion
     device.vm.provision :shell , :inline => <<-udev_rule
```

File: docs/developer/integration_tests/virtual_machines.asciidoc [hunk 3]
```diff
@@ -49,20 +49,22 @@ link:https://docs.cumulusnetworks.com/version/cumulus-linux-37/Layer-1-and-Switc
...
 
 `node01` is a Debian Buster virtual machine used as node to test wired PacketFence
 features like 802.1X, MAC authentification, registration, isolation and inline
 scenarios.
 
-This machine has an Internet access only during its first boot to emulate a
-node connected to a switch controlled by PacketFence. After the reboot, this machine is only
+`node02` is identical to `node01` but is only used to test inline L3 features of PacketFence.
+
+Machine have an Internet access only during its first boot to emulate a
```
Comment (JeGoi) [inline docs/developer/integration_tests/virtual_machines.asciidoc]:
```suggestion
These machines have an Internet access only during its first boot to emulate a
```

File: docs/developer/integration_tests/virtual_machines.asciidoc [hunk 4]
```diff
@@ -49,20 +49,22 @@ link:https://docs.cumulusnetworks.com/version/cumulus-linux-37/Layer-1-and-Switc
...
+node connected to a switch controlled by PacketFence. After the reboot, machine are only
```
Comment (JeGoi) [inline docs/developer/integration_tests/virtual_machines.asciidoc]:
```suggestion
node connected to a switch controlled by PacketFence. After the reboot, machines are only
```

File: addons/vagrant/helper_scripts/config_node02.sh [hunk 5]
```diff
@@ -0,0 +1,55 @@
...
+echo -e "iface eth0 inet dhcp\n\n" >> /etc/network/interfaces
+
+# Make DHCP Try Over and Over Again
+echo "retry 1;" >> /etc/dhcp/dhclient.conf
+
+# Other stuff
+ping 8.8.8.8 -c2
+if [ "$?" == "0" ]; then
+    apt-get update -qy && apt-get install gnupg -qy
+    apt-get update -qy
```
Comment (JeGoi) [inline addons/vagrant/helper_scripts/config_node02.sh]:
```suggestion

```

Comment (JeGoi) [reply addons/vagrant/helper_scripts/config_node02.sh]: Not sure that one is useful

**Review by JeGoi — COMMENTED** [review]
Only few modifications

**Comment by nqb** [comment]
@JeGoi, thanks for you review.

I applied your suggestions in latest commit and generate to node01.

**Comment by nqb** [comment]
@JeGoi, could you do another review ?

Thanks.

**Comment by JeGoi** [comment]
It looks that chmod has changed for addons/vagrant/helper_scripts/config_node01.sh from 0755 to 0644.

File: addons/vagrant/cumulus/Vagrantfile [hunk 6]
```diff
@@ -361,4 +386,110 @@ vagrant_interface_rule
...
+    end
+
+    # Run the Config specified in the Node Attributes
+    device.vm.provision :shell , privileged: false, :inline => 'echo "$(whoami)" > /tmp/normal_user'
+    device.vm.provision "config_node", type: "shell", path: "./helper_scripts/config_node02.sh"
+
+
+    # Install Rules for the interface re-map
+    device.vm.provision :shell , :inline => <<-delete_udev_directory
+if [ -d "/etc/udev/rules.d/70-persistent-net.rules" ]; then
```
Comment (JeGoi) [inline addons/vagrant/cumulus/Vagrantfile:456]:
```suggestion
    if [ -d "/etc/udev/rules.d/70-persistent-net.rules" ]; then
```

File: addons/vagrant/cumulus/Vagrantfile [hunk 7]
```diff
@@ -361,4 +386,110 @@ vagrant_interface_rule
...
+    rm -rfv /etc/udev/rules.d/70-persistent-net.rules &> /dev/null
```
Comment (JeGoi) [inline addons/vagrant/cumulus/Vagrantfile:457]:
```suggestion
        rm -rfv /etc/udev/rules.d/70-persistent-net.rules &> /dev/null
```

File: addons/vagrant/cumulus/Vagrantfile [hunk 8]
```diff
@@ -361,4 +386,110 @@ vagrant_interface_rule
...
+fi
```
Comment (JeGoi) [inline addons/vagrant/cumulus/Vagrantfile:458]:
```suggestion
    fi
```

File: addons/vagrant/cumulus/Vagrantfile [hunk 9]
```diff
@@ -361,4 +386,110 @@ vagrant_interface_rule
...
+delete_udev_directory
```
Comment (JeGoi) [inline addons/vagrant/cumulus/Vagrantfile:459]:
```suggestion
    delete_udev_directory
```

**Review by JeGoi — COMMENTED** [review]
Only few tabs and the chmod
Looks OK for me but did not test it yet.

**Comment by nqb** [comment]
I'm not sure indentation using a herescript is supported like you proposed.

I removed a useless `rm` like you reported last time and fixed chmod.

**Review by JeGoi — APPROVED** [review]

**Pull request merged**